lib/app/dispatcher.rb

```diff
@@ -82,9 +82,11 @@ def start_polling_jobs!
 
           pool.post do
             send_heartbeat(connector_id, service_type)
```

Comment:
I think it's also worth to separate actual heartbeat from the configuration change - probably for a different PR